src/app/components/Citations/APACitation.jsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+
+class APACitation extends React.Component {
+  static getInitials(name) {
+    return name.trim().split(/\s+/).map(i => `${i.substr(0, 1).toUpperCase()}.`).join(' ');
+  }
+
+  componentWillMount() {
+    this.formatCitationData();
+  }
+
+  formatCitationData() {
+    console.log(this.props)
+    this.title = this.props.subTitle ? `${this.props.title}: ${this.props.subTitle}` : this.props.title;
+
+    this.authors = this.formatAgentNames('authors');
+    const rawEditors = this.formatAgentNames('editors', this.props.agents.authors.length > 0);
+    this.editors = `${rawEditors}, (Ed${this.props.agents.editors.length === 1 ? '.' : 's.'})`;
+    this.translators = `${this.formatAgentNames('translators', true)}, Trans.`;
+    this.illustrators = `${this.formatAgentNames('illustrators', true)}, Illus.`;
+
+    this.pubYear = this.props.publicationYear ? ` (${this.props.publicationYear})` : '';
+    this.publishers = this.props.agents.publishers.join(', ');
+
+    this.edition = this.props.edition ? ` (${this.props.edition})` : '';
+    this.volume = this.props.volume ? ` (${this.props.volume})` : '';
+
+    this.link = this.props.sourceLink ? <a href={`https://${this.props.sourceLink}`}>{this.props.sourceLink}</a> : '';
+
+    this.govReportStatus = this.props.isGovernmentDoc;
+  }
+
+
+  formatAgentNames(agentType, nameFirst = false) {
```

Comment:
A tip:  It'll be a lot easier to test functions like these if you move them out of the component (so you won't have to simulate the component and passing props) 
(so pass `this.props.agents` into the function)

Reply:
I tried a few different ways of organizing this logic, and it feels like it needs to be here since this is so closely tied to displaying the APA format. I'll keep that in mind though if I can't get it to test cleanly.